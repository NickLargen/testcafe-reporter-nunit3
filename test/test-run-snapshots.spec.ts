import * as mockDate from 'jest-date-mock';
import { createReport } from './utils/create-report';
import { sampleCalls, demoFruitCalls, CallSequence } from './utils/reporter-test-calls';

// Don't put the entire snapshot in quotes, forcing quote content to be escaped
expect.addSnapshotSerializer({ serialize: (val) => val, test: (val) => typeof val === 'string' });

beforeEach(() => {
  mockDate.advanceTo('2077-03-14T03:14:15.926Z');
});

afterEach(() => {
  mockDate.clear();
});

describe('Regression testing ', () => {
  function expectSnapshotMatch(calls: CallSequence) {
    const actual = createReport(calls)
      .replace(/<filePath>.*?<\/filePath>/g, '<filePath>FILE_PATH</filePath>')
      .replace(/\((.+?)(:\d+:\d+)\)/g, ($0, $1, $2) => `(FILE_PATH${$2})`);
    expect(actual).toMatchSnapshot();
  }

  it('should match for the example data used by the official TestCafe reporters', function () {
    expectSnapshotMatch(sampleCalls);
  });

  it('should match for data copied from a test run with quarantine mode on', function () {
    expectSnapshotMatch(demoFruitCalls);
  });
});
