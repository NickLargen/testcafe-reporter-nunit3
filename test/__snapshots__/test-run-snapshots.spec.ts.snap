// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regression testing  should match for data copied from a test run with quarantine mode on 1`] = `
<?xml version="1.0" encoding="UTF-8" ?>
<test-run id="2" name="TestCafe Tests" start-time="1970-03-14T15:27:12.448Z" end-time="1970-03-14T15:28:03.175Z">
    <test-suite type="Assembly" name="Banana" start-time="2077-03-14T03:14:15.926Z" end-time="2077-03-14T03:14:15.926Z">
        <test-case name="Split" fullname="Banana - Split" result="Failed" duration="17.2">
            <failure>
                <message>
                    <![CDATA[Run 1: Failed
Run 2: Failed
Run 3: Failed]]>
                </message>
                <stack-trace>
                    <![CDATA[-- AssertionError: expected '41' to deeply equal '7'
⠀
   Browser: Chrome 80.0.3987.132 / Windows 10
   Screenshot: C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-1\\errors\\1_run-3_Chrome.png]]>
                </stack-trace>
            </failure>
            <attachments>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-1\\errors\\1_run-1_Chrome.png</filePath>
                </attachment>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-1\\errors\\1_run-2_Chrome.png</filePath>
                </attachment>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-1\\errors\\1_run-3_Chrome.png</filePath>
                </attachment>
            </attachments>
        </test-case>
        <test-case name="Costume" fullname="Banana - Costume" result="Failed" duration="15.978">
            <failure>
                <message>
                    <![CDATA[Run 1: Failed
Run 2: Failed
Run 3: Failed]]>
                </message>
                <stack-trace>
                    <![CDATA[-- AssertionError: expected false to be truthy
⠀
   Browser: Chrome 80.0.3987.132 / Windows 10
⠀
    ⠀  3 |export let testCallsiteVariableLineNumberLength: any, testCallsiteAllLinesTwoDigitNumbers: any;
    ⠀  4 |
    ⠀  5 |// ----- Avoid changing the contents of this file, they constitute mock failure locations ---- \\\\
    ⠀  6 |
    ⠀  7 |function functionThatThrowsUp() {
    >  8 |  throw new Error('Hey ya!');
    ⠀  9 |}
   ⠀  10 |
   ⠀  11 |try {
   ⠀  12 |  functionThatThrowsUp();
   ⠀  13 |} catch (err) {
⠀
      at functionThatThrowsUp (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)
      at Runtime.requireModuleOrMock (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)]]>
                </stack-trace>
            </failure>
            <attachments>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-2\\errors\\1_run-1_Chrome.png</filePath>
                </attachment>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-2\\errors\\1_run-2_Chrome.png</filePath>
                </attachment>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-2\\errors\\1_run-3_Chrome.png</filePath>
                </attachment>
            </attachments>
        </test-case>
    </test-suite>
    <test-suite type="Assembly" name="Pear" start-time="2077-03-14T03:14:15.926Z" end-time="2077-03-14T03:14:15.926Z">
        <test-case name="Fast" fullname="Pear - Fast" result="Passed" duration="0.716">
        </test-case>
        <test-case name="Faster" fullname="Pear - Faster" result="Failed" duration="16.582">
            <failure>
                <message>
                    <![CDATA[Run 1: Failed
Run 2: Failed
Run 3: Failed]]>
                </message>
                <stack-trace>
                    <![CDATA[-- AssertionError: expected '41' to deeply equal '5'
⠀
   Browser: Chrome 80.0.3987.132 / Windows 10
   Screenshot: C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-4\\errors\\1_run-3_Chrome.png]]>
                </stack-trace>
            </failure>
            <attachments>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-4\\errors\\1_run-1_Chrome.png</filePath>
                </attachment>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-4\\errors\\1_run-2_Chrome.png</filePath>
                </attachment>
                <attachment>
                    <filePath>C:\\testcafe-demo\\screenshots\\1970-03-14_11-27-11\\test-4\\errors\\1_run-3_Chrome.png</filePath>
                </attachment>
            </attachments>
        </test-case>
    </test-suite>
</test-run>
`;

exports[`Regression testing  should match for the example data used by the official TestCafe reporters 1`] = `
<?xml version="1.0" encoding="UTF-8" ?>
<test-run id="2" name="TestCafe Tests" start-time="1970-01-01T00:00:00.000Z" end-time="1970-01-01T00:15:25.000Z">
    <test-suite type="Assembly" name="First fixture" start-time="2077-03-14T03:14:15.926Z" end-time="2077-03-14T03:14:15.926Z">
        <test-case name="First test in first fixture" fullname="First fixture - First test in first fixture" result="Inconclusive" duration="74">
            <attachments>
                <attachment>
                    <filePath>C:/screenshots/1445437598847/1.png</filePath>
                </attachment>
                <attachment>
                    <filePath>C:/screenshots/1445437598847/2.png</filePath>
                </attachment>
            </attachments>
        </test-case>
        <test-case name="Second test in first fixture" fullname="First fixture - Second test in first fixture" result="Failed" duration="74">
            <failure>
                <message>
                    <![CDATA[Some error
]]>
                </message>
                <stack-trace>
                    <![CDATA[-- A JavaScript error occurred on "http://example.org".
   Repeat test actions in the browser and check the console for errors.
   If you see this error, it means that the tested website caused it. You can fix it or disable tracking JavaScript errors in TestCafe. To do the latter, enable the "--skip-js-errors" option.
   If this error does not occur, please write a new issue at:
   "https://github.com/DevExpress/testcafe/issues/new?template=bug-report.md".
⠀
   JavaScript error details:
   Some error
⠀
   Browser: Chrome 41.0.2227 / Mac OS X 10.10.1
⠀
    ⠀  3 |export let testCallsiteVariableLineNumberLength: any, testCallsiteAllLinesTwoDigitNumbers: any;
    ⠀  4 |
    ⠀  5 |// ----- Avoid changing the contents of this file, they constitute mock failure locations ---- \\\\
    ⠀  6 |
    ⠀  7 |function functionThatThrowsUp() {
    >  8 |  throw new Error('Hey ya!');
    ⠀  9 |}
   ⠀  10 |
   ⠀  11 |try {
   ⠀  12 |  functionThatThrowsUp();
   ⠀  13 |} catch (err) {
⠀
      at functionThatThrowsUp (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)
      at Runtime.requireModuleOrMock (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)
⠀
-- The specified selector does not match any element in the DOM tree.
⠀
      | Lorem ipsum dolor sit amet, consectetur adipiscing...)
    > |   one
      |   two
      |   three
⠀
   Browser: Firefox 47 / Mac OS X 10.10.1
⠀
   ⠀  13 |} catch (err) {
   ⠀  14 |  testCallsiteVariableLineNumberLength = createCallsiteRecord({ forError: err });
   ⠀  15 |}
   ⠀  16 |
   ⠀  17 |const functionThatThrowsDown = () => {
   ⠀> 18 |  throw Error('Wilson!');
   ⠀  19 |};
   ⠀  20 |
   ⠀  21 |try {
   ⠀  22 |  functionThatThrowsDown();
   ⠀  23 |} catch (err) {
⠀
      at functionThatThrowsDown (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)
      at Runtime.requireModuleOrMock (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)]]>
                </stack-trace>
            </failure>
        </test-case>
        <test-case name="Third test in first fixture" fullname="First fixture - Third test in first fixture" result="Passed" duration="74">
        </test-case>
    </test-suite>
    <test-suite type="Assembly" name="Second fixture" start-time="2077-03-14T03:14:15.926Z" end-time="2077-03-14T03:14:15.926Z">
        <test-case name="First test in second fixture" fullname="Second fixture - First test in second fixture" result="Passed" duration="74">
        </test-case>
        <test-case name="Second test in second fixture" fullname="Second fixture - Second test in second fixture" result="Passed" duration="74">
        </test-case>
        <test-case name="Third test in second fixture" fullname="Second fixture - Third test in second fixture" result="Passed" duration="0">
        </test-case>
    </test-suite>
    <test-suite type="Assembly" name="Third fixture" start-time="2077-03-14T03:14:15.926Z" end-time="2077-03-14T03:14:15.926Z">
        <test-case name="First test in third fixture" fullname="Third fixture - First test in third fixture" result="Failed" duration="74">
            <failure>
                <message>
                    <![CDATA[]]>
                </message>
                <stack-trace>
                    <![CDATA[-- - Error in test.before hook -
   The specified selector does not match any element in the DOM tree.
⠀
      | Lorem ipsum dolor sit amet, consectetur adipiscing...)
    > |   one
      |   two
      |   three
⠀
   Browser: Firefox 47 / Mac OS X 10.10.1
⠀
   ⠀  13 |} catch (err) {
   ⠀  14 |  testCallsiteVariableLineNumberLength = createCallsiteRecord({ forError: err });
   ⠀  15 |}
   ⠀  16 |
   ⠀  17 |const functionThatThrowsDown = () => {
   ⠀> 18 |  throw Error('Wilson!');
   ⠀  19 |};
   ⠀  20 |
   ⠀  21 |try {
   ⠀  22 |  functionThatThrowsDown();
   ⠀  23 |} catch (err) {
⠀
      at functionThatThrowsDown (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)
      at Runtime.requireModuleOrMock (some-file:1:1)
      at Object.<anonymous> (some-file:1:1)
      at Runtime._execModule (some-file:1:1)
      at Runtime._loadModule (some-file:1:1)
      at Runtime.requireModule (some-file:1:1)]]>
                </stack-trace>
            </failure>
        </test-case>
    </test-suite>
</test-run>
`;
